### ISSUE TITLE
Unable to preload nested association

### ISSUE THREAD

**Issue by dannyongtey** [djezzzl/n1_loader#45]
Hi there! First of all, thanks for the wonderful gem! 

This is my first ever bug report in any open source repository, so do let me know if there are things I have done wrongly. 🙏 

I noticed that nested preloading does not work, if the nested association's key is from an n1_loader loaded association. Here's a reproducible code example (Adapted from issue 36): 

```
require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem 'activerecord', '~> 7.0.0' # Can be 6 also
  gem 'n1_loader', '1.7.3', require: 'n1_loader/active_record'
  gem 'sqlite3'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'

ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
ActiveRecord::Base.logger = Logger.new($stdout)

ActiveRecord::Schema.define do
  create_table :authors, force: true do |t|
  end

  create_table :posts, force: true do |t|
    t.integer :author_id
    t.text :body
  end

  create_table :comments, force: true do |t|
    t.integer :post_id
  end
end

class Author < ActiveRecord::Base
  has_many :posts
  n1_optimized :posts_with_n1_loader do |authors|
    authors.each do |author|
      fulfill(author, author.posts) # or author.posts.to_a
    end
  end
end

class Post < ActiveRecord::Base
  has_many :comments
  belongs_to :author
end

class Comment < ActiveRecord::Base
  belongs_to :post
end

class BugTest < Minitest::Test
  def test_nested_preloading
    author = Author.create!
    post = Post.create!(author: author, body: "test")
    comment = Comment.create!(post: post)
    
    author = Author.includes(posts_with_n1_loader: :comments)
    assert_equal author.first.posts_with_n1_loader, author.first.posts
  end
end
```

It returns 
```
NoMethodError: undefined method `_reflect_on_association' for Post::ActiveRecord_Associations_CollectionProxy:Class
```

However, by including the following line, the issue seems to go away:
```
N1Loader::Loader.define_method :preloaded_records do
  @preloaded_records ||= loaded.values.flatten
end
```

The `posts_with_n1_loader` association is redundant in this case, but in the case of other custom logic, it seems to also fail.

It seems that the variable preloaded_records is a nested array, and hence when deciding the preloading logic, it doesn't detect the value as one loaded from n1_loader, and runs the original Rails preloading strategy.

I am not an expert in Rails or Ruby, so let me know if there's anything I might have overlooked! 

Thanks a lot!

**Comment by dannyongtey** [comment]
Hi @djezzzl,

I'm wondering if you have managed to have a look at this yet? 

Thanks in advance for your time - let me know if there's anything I could do to assist in any way :)

**Comment by dannyongtey** [comment]
Hi @djezzzl , 

Thank you for getting back! I have prepared a PR, let me know if this is the right fix :)

**Comment by djezzzl** [comment]
Thank you! Merged and released as 1.7.4! Have a great time!

**Comment by dannyongtey** [comment]
Thank you! Hope you have a great time too <3
